src/ojo/agent/OjosDatabaseHandler.cc

```diff
@@ -0,0 +1,354 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+/**
+ * @file
+ * @brief Data base handler for OJO
+ */
+
+#include "OjosDatabaseHandler.hpp"
+
+using namespace fo;
+using namespace std;
+
+/**
+ * Default constructor for OjosDatabaseHandler
+ * @param dbManager DBManager to be used
+ */
+OjosDatabaseHandler::OjosDatabaseHandler(DbManager dbManager) :
+    fo::AgentDatabaseHandler(dbManager)
+{
+}
+
+/**
+ * Get a vector of all file id for a given upload id.
+ * @param uploadId Upload ID to be queried
+ * @return List of all pfiles for the given upload
+ */
+vector<unsigned long> OjosDatabaseHandler::queryFileIdsForUpload(int uploadId)
+{
+  return queryFileIdsVectorForUpload(uploadId);
+}
+
+/**
+ * Get a vector of all file id for a given upload id which are not scanned by the given agentId.
+ * @param uploadId Upload ID to be queried
+ * @param agentId  ID of the agent
+ * @return List of all pfiles for the given upload
+ */
+vector<unsigned long> OjosDatabaseHandler::queryFileIdsForScan(int uploadId, int agentId)
+{
+  string uploadtreeTableName = queryUploadTreeTableName(uploadId);
+
+  QueryResult queryResult = dbManager.execPrepared(
+    fo_dbManager_PrepareStamement(dbManager.getStruct_dbManager(),
+      ("pfileForUploadFilterAgent" + uploadtreeTableName).c_str(),
+      ("SELECT distinct(ut.pfile_fk) FROM " + uploadtreeTableName + " AS ut "
+      "LEFT JOIN license_file AS lf ON ut.pfile_fk = lf.pfile_fk "
+      "AND lf.agent_fk = $2 WHERE lf.pfile_fk IS NULL "
+      "AND ut.upload_fk = $1 AND (ut.ufile_mode&x'3C000000'::int)=0;").c_str(),
+      int, int),
+    uploadId, agentId);
+
+  return queryResult.getSimpleResults(0, fo::stringToUnsignedLong);
+}
+
+/**
+ * Spawn a new DbManager object.
+ *
+ * Used to create new objects for threads.
+ * @return DbManager object for threads.
+ */
+OjosDatabaseHandler OjosDatabaseHandler::spawn() const
+{
+  DbManager spawnedDbMan(dbManager.spawn());
+  return OjosDatabaseHandler(spawnedDbMan);
+}
+
+/**
+ * @brief Save findings to the database if agent was called by scheduler
+ * @param entry The entry to be made
+ * @return fl_pk on success, -1 on failure
+ */
+unsigned long OjosDatabaseHandler::saveLicenseToDatabase(
+    OjoDatabaseEntry &entry) const
+{
+  QueryResult queryResult = dbManager.execPrepared(
+    fo_dbManager_PrepareStamement(dbManager.getStruct_dbManager(),
+      "ojoInsertLicense",
+      "INSERT INTO license_file"
+      "(rf_fk, agent_fk, pfile_fk, rf_match_pct)"
+      " VALUES($1,$2,$3,100) RETURNING fl_pk",
+      long, long, long),
```

Comment:
100 is not require as Ojo don't find any percentage

Reply:
Removed, please check.